### PR TITLE
Update BI connectors and drivers readme files

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury Fridlyand <yuryf@bitquilltech.com>

* Update `readme`s
* Add connectors' binaries
* Add connectors' download links
* Add and update connectors' installation and usage instructions
* Update screenshots to remove ODFE references
* Add drivers' download links
* Reorganize connectors' file tree
* Update GHA workflow

## Resolves
https://github.com/opensearch-project/sql/issues/605
https://github.com/opensearch-project/sql/issues/607
https://github.com/opensearch-project/sql/issues/629
https://github.com/opensearch-project/sql/issues/439
https://github.com/opensearch-project/sql/issues/340
https://github.com/opensearch-project/sql/issues/246
https://github.com/opensearch-project/sql/issues/257

### CODE REVIEW

File: bi-connectors/PowerBIConnector/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Connector Download
+
+The most recent build could be donwloaded from an automated CI workflow: [link](https://github.com/opensearch-project/sql/actions/workflows/bi-connectors.yml).
```
Comment (acarbonetto) [inline bi-connectors/PowerBIConnector/README.md:3]: `The most recent build could be donwloaded from an automated CI workflow`
to:
`The Power BI connector is available to be downloaded from the automated CI workflow`

File: bi-connectors/TableauConnector/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+## Connector Download
+
+The most recent build could be donwloaded from an automated CI workflow: [link](https://github.com/opensearch-project/sql/actions/workflows/bi-connectors.yml).
```
Comment (acarbonetto) [inline bi-connectors/TableauConnector/README.md:3]: Recommend changing from: 
`The most recent build could be donwloaded from an automated CI workflow`
to:
`The Tableau connector is available to be downloaded from the automated CI workflow`

File: sql-jdbc/README.md [hunk 3]
```diff
@@ -6,6 +6,15 @@ This is the driver for JDBC connectivity to a cluster running with OpenSearch SQ
...
 The driver is compatible with JDBC 4.2 specification and requires a minimum of Java 8.
 
+## BI Tool Connectors
+
+* [Power BI Desktop](../bi-connectors/PowerBIConnector/README.md)
+* [Tableau Desktop](../bi-connectors/TableauConnector/README.md)
+
+## Download and Installation
+
+The driver is available for download from an [automated CI workflow](https://github.com/opensearch-project/sql/actions/workflows/sql-jdbc-test-and-build-workflow.yml) and from [Artifacts page](https://opensearch.org/artifacts) on OpenSearch.org at the very bottom.
```
Comment (acarbonetto) [inline sql-jdbc/README.md:16]: Can you swap the ordering?  We should recommend that the JDBC driver be downloaded from the artifacts directory first.

Comment (acarbonetto) [reply sql-jdbc/README.md:16]: Also, please remove 'an' from: `The driver is available for download from an`

File: sql-odbc/README.md [hunk 4]
```diff
@@ -9,19 +9,20 @@ The driver is compatible with ODBC 3.51.
...
-* **Power BI Desktop**: [`OpenSearch.mez`](./src/PowerBIConnector/bin/Release/OpenSearch.mez)
-* **Tableau**: [`opensearch_sql_odbc.taco`](./src/TableauConnector/opensearch_sql_odbc/opensearch_sql_odbc.taco)
+* [Power BI Desktop](../bi-connectors/PowerBIConnector/README.md)
+* [Tableau Desktop](../bi-connectors/TableauConnector/README.md)
 
-## Installing the Driver
+## Download and Installation the Driver
 
 You can use the installers generated as part of the most recent release.
+They are available for download from an [automated CI workflow](https://github.com/opensearch-project/sql/actions/workflows/sql-odbc-main.yml) and from [Artifacts page](https://opensearch.org/artifacts) on OpenSearch.org at the very bottom.
```
Comment (acarbonetto) [inline sql-odbc/README.md:25]: Same. 
Please swap the order. 
Remove 'an' from `They are available for download from an`.

**Review by acarbonetto — CHANGES_REQUESTED** [review]

**Review by acarbonetto — APPROVED** [review]

File: sql-jdbc/README.md [hunk 5]
```diff
@@ -6,6 +6,15 @@ This is the driver for JDBC connectivity to a cluster running with OpenSearch SQ
...
 The driver is compatible with JDBC 4.2 specification and requires a minimum of Java 8.
 
+## BI Tool Connectors
+
+* [Power BI Desktop](../bi-connectors/PowerBIConnector/README.md)
+* [Tableau Desktop](../bi-connectors/TableauConnector/README.md)
+
+## Download and Installation
+
+The driver is available for download from [Artifacts page](https://opensearch.org/artifacts) on OpenSearch.org at the very bottom from and [automated CI workflow](https://github.com/opensearch-project/sql/actions/workflows/sql-jdbc-test-and-build-workflow.yml).
```
Comment (acarbonetto) [inline sql-jdbc/README.md:16]: switch `from and` to `and from` at the end of the line

**Review by MaxKsyunz — APPROVED** [review]

File: sql-odbc/README.md [hunk 6]
```diff
@@ -9,19 +9,20 @@ The driver is compatible with ODBC 3.51.
...
-* **Power BI Desktop**: [`OpenSearch.mez`](./src/PowerBIConnector/bin/Release/OpenSearch.mez)
-* **Tableau**: [`opensearch_sql_odbc.taco`](./src/TableauConnector/opensearch_sql_odbc/opensearch_sql_odbc.taco)
+* [Power BI Desktop](../bi-connectors/PowerBIConnector/README.md)
+* [Tableau Desktop](../bi-connectors/TableauConnector/README.md)
 
-## Installing the Driver
+## Download and Installation the Driver
 
 You can use the installers generated as part of the most recent release.
+They are available for download from [Artifacts page](https://opensearch.org/artifacts) on OpenSearch.org at the very bottom from and [automated CI workflow](https://github.com/opensearch-project/sql/actions/workflows/sql-odbc-main.yml).
```
Comment (acarbonetto) [inline sql-odbc/README.md:25]: at the end of the line: switch `from and` to `and from`

**Review by forestmvey — APPROVED** [review]

**Pull request merged**